react/features/subtitles/middleware.js

```diff
@@ -41,15 +50,33 @@ const REMOVE_AFTER_MS = 3000;
  * @returns {Function}
  */
 MiddlewareRegistry.register(store => next => action => {
-
     switch (action.type) {
     case ENDPOINT_MESSAGE_RECEIVED:
         return _endpointMessageReceived(store, next, action);
+    case TOGGLE_REQUESTING_SUBTITLES:
+        _requestingSubtitlesToggled(store);
+        break;
     }
 
     return next(action);
 });
 
+/**
+ * Toggle the local property 'requestingTranscription'. This will cause Jicofo
+ * and Jigasi to decide whether the transcriber needs to be in the room.
+ *
+ * @param {Store} store - The redux store.
+ * @private
+ * @returns {void}
+ */
+function _requestingSubtitlesToggled({ getState }) {
+    const { _requestingSubtitles } = getState()['features/subtitles'];
+    const { conference } = getState()['features/base/conference'];
+
+    conference.setLocalParticipantProperty(P_NAME_REQUESTING_TRANSCRIPTION,
+        !_requestingSubtitles);
+}
```

Comment:
You might want to use a subscriber instead to react to state changes. Up to you though; using middleware is already in line with how most of everything else is set up.